lib/streamlit/watcher/util.py

```diff
@@ -54,6 +63,12 @@ def calc_md5_with_blocking_retries(path: str) -> str:
     return md5.hexdigest()
 
 
+def path_modification_time(path: str, allow_nonexistent: bool = False) -> float:
+    if allow_nonexistent and not os.path.exists(path):
+        return 0.0
+    return os.stat(path).st_mtime
+
+
```

Comment:
I don't quite understand what this function is doing. What happens if the path is non-existent and `allow_nonexistent` is False? Does it raise an expected exception? (Could we add a docstring here?)

Reply:
done!